test/filters/test_grid_filter.cpp

```diff
@@ -0,0 +1,193 @@
+/*
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Point CLoud Library (PCL) - www.pointclouds.org
+ * Copyright (c) 2020-, Open Perception
+ *
+ * All rights reserved
+ */
+
+#include <pcl/filters/experimental/grid_filter_base.h>
+#include <pcl/io/pcd_io.h>
+#include <pcl/test/gtest.h>
+#include <pcl/pcl_tests.h>
+#include <pcl/point_types.h>
+
+#include <boost/optional.hpp> // std::optional for C++17
+
+#include <unordered_map>
+
+using namespace pcl;
+using namespace pcl::test;
+
+// Grid structure with necessary declarations for input to GridFilterBase
+
+struct EmptyVoxel {
+  bool voxel_info = false;
+};
+
+// hash map grid
+template <typename PointT>
+struct EmptyMapStruct {
+  using PointCloud = pcl::PointCloud<PointT>;
+  using Grid = std::unordered_map<std::size_t, EmptyVoxel>;
+
+  EmptyMapStruct() : filter_name_("empty_map") {}
+
+protected:
+  // accessing GridFilterBase
+  const experimental::GridFilterBase<EmptyMapStruct>*
+  getDerived()
+  {
+    return static_cast<const experimental::GridFilterBase<EmptyMapStruct>*>(this);
+  }
+
+  bool
+  setUp()
+  {
+    // test passed or failed setUp
+    return getDerived()->getDownsampleAllData();
```

Comment:
What is this for? If this is being used to do double duty, let's use another variable instead

Reply:
Done

Reply:
Done, I declared new variables in the testing class for the job